wots.c

```diff
@@ -109,6 +111,78 @@ static void chain_lengths(const xmss_params *params,
  *
  * Writes the computed public key to 'pk'.
  */
+
+typedef struct wots_pkgen_args
+{
+    const xmss_params *params;
+    uint32_t *addr;
+    uint32_t start;
+    uint32_t end;
+    uint32_t num;
+    const unsigned char *pub_seed;
+    unsigned char *pk;
+    uint16_t padding16;
+    uint32_t padding32;
+} wots_pkgen_args_t;
+
+static void *wots_pkgen_sub(void *vars)
+{
+    wots_pkgen_args_t *args = vars;
+    for (uint32_t i = args->start; i < args->end; i++)
+    {
+        set_chain_addr(args->addr, i);
+        gen_chain(args->params, args->pk + i * (args->params->n), args->pk + i * (args->params->n),
+                  0, args->params->wots_w - 1, args->pub_seed, args->addr);
+    }
+    return NULL;
+}
+
+void wots_pkgen_mp(const xmss_params *params,
+                unsigned char *pk, const unsigned char *seed,
+                const unsigned char *pub_seed, uint32_t addr[8])
```

Comment:
The use of "naked" constants makes me nervous. Can you give a name for this value 8, perhaps as a macro, and use it instead?

Reply:
The naked constants comes from the original implementation, and it only use for providing information to the hash chain. This fixed size information is used in regard of XMSS or XMSS^MT. 
I think we can keep it because I don't have better idea to replace it with a macro, and it does not need to be changed in anytime soon.